File: chillvie/src/service/MovieService.js
import axios from "axios";

export async function getMovieIsShowing(){
    const res = await axios.get("http://localhost:8080/api/movie/showing");
    return res.data;
}

export async function getListShowingMovie(name){
    const res = await axios.get("http://localhost:8080/api/movie/showing-movie?name="+name);
    return res.data;
}

export async function getListUpcomingMovie(name){
    const res = await axios.get("http://localhost:8080/api/movie/upcoming-movie?name="+name);
    return res.data;
}

export async function getMovieById(id){
    const res = await axios.get("http://localhost:8080/api/movie/movie?id="+id);
    return res.data;
}

File: back-end-chillvie/src/main/java/com/example/backendchillvie/service/MovieService.java
package com.example.backendchillvie.service;

import com.example.backendchillvie.model.Movie;
import com.example.backendchillvie.projection.IMovieProjection;
import com.example.backendchillvie.repository.IMovieRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class MovieService implements IMovieService{
    @Autowired
    private IMovieRepository iMovieRepository;
    @Override
    public List<Movie> getMovieIsShowing() {
        return iMovieRepository.getMoviesByIsShowingIsTrueAndFlagDeletedIsFalse();
    }

    @Override
    public List<IMovieProjection> getListUpcomingMovie(String name) {
        return iMovieRepository.getListUpcomingMovie(name);
    }

    @Override
    public List<IMovieProjection> getListShowingMovie(String name) {
        return iMovieRepository.getListShowingMovie(name);
    }

    @Override
    public Optional<Movie> getMovieById(Long id) {
        return iMovieRepository.getMoviesByIdAndFlagDeletedIsFalse(id);
    }
}
